### ISSUE TITLE
Voice Call Audio Delay 

### ISSUE THREAD

**Issue by NemtsevSergey** [sipsorcery/sipsorcery#273]
Hello, i noticed that when i call from "operator 1" to "operator 2" - and say "hello" in "operator 1" - i can hear "hello" after at least 2-3 seconds in "operator 2". Test was with SIPSorcery.SoftPhone.
Then i check this with PhonerLite and delay was about 1 second or less. 
What could be the reason for such a delay?

**Comment by sipsorcery** [comment]
Can you take a wireshark capture of the RTP, preferably from the end of the call suffering the audio issues, and attach it or email it to me at `aaron@sipsorcery.com`.

This issue #234 is vaguely similar and in the end the audio issues seemed to be because the PC running the softphone app was too busy.

**Comment by NemtsevSergey** [comment]
Hello, i try to record data during call.

**Comment by NemtsevSergey** [comment]
Wireshark work

[Capture.zip](https://github.com/sipsorcery/sipsorcery/files/5031880/Capture.zip)

**Comment by sipsorcery** [comment]
What's the IP address of the machine you ran the capture on? And whereabouts is the sipsorcery softphone app running?

`10.1.0.124` or `10.1.21.116`?

In the RTP analysis graphs below the RTP being received on `10.1.0.124` is very choppy.

![rtp_analysis_124_116](https://user-images.githubusercontent.com/197660/89496724-cc4d7400-d7b2-11ea-8ab7-516cd3e5fa2f.png)

![rtp_analysis_116_124](https://user-images.githubusercontent.com/197660/89496727-ceafce00-d7b2-11ea-8f96-12920f6aae8b.png)

**Comment by NemtsevSergey** [comment]
My PC is 10.1.21.116.

**Comment by sipsorcery** [comment]
Are you using the .NET Framework or .NET Core?

The spikes in the graph are very consistent which points to something in the softphone app or on your PC hogging the CPU at regular intervals. Run the Resource Monitor while placing the call and check the CPU activity.

**Comment by NemtsevSergey** [comment]
We use .net core 3.1.  Ok we will check CPU

**Comment by NemtsevSergey** [comment]
Hello, CPU activity is not higth during call. I capture new activity. 
![1](https://user-images.githubusercontent.com/36694542/89841151-90048400-dbb5-11ea-808f-1b78ac11f5fd.PNG)
[NewCapture.zip](https://github.com/sipsorcery/sipsorcery/files/5053949/NewCapture.zip)

**Comment by sipsorcery** [comment]
OK CPU usage looks very low.

What's the source of your audio? A microphone? Are you using the `SIPSorcery.SoftPhone` example as is so that the audio is supplied by `NAudio`?

**Comment by NemtsevSergey** [comment]
Yes, source is from SIPSorcery.Media library.

**Comment by NemtsevSergey** [comment]
После очередного анализа выявили, что на записи в АТС - голоса без задержки. То есть задержка идёт именно при воспроизведении звука в программе.

**Comment by sipsorcery** [comment]
I had a closer look at your packet capture and I actually think there isn't any major problem with it. Yes there are spikes in the time between the audio packets but they are within 10ms and the max jitter is 4.59ms which is way too small for the human ear to detect. So based on your packet capture I don't think the 1s delay is likely to be coming from your PC or the `SIPSorcery Softphone` app.

Are you able to get a packet capture at the Asterisk end of the call? It would be very useful to compare captures of the same call on the source and destination.

![jitter](https://user-images.githubusercontent.com/197660/90015711-de428180-dca0-11ea-875b-1c5e48f63e4c.png)

I also ran a test on my Windows 10 PC by placing a call between the `SIPSorcery Softphone` and the `MicroSIP Softphone` both on the same PC. The maximum jitter I got from the `SIPSorcery Softphone` was 9.13ms which is worse than yours but still way under what a human ear can detect.

In the jitter graph from my test the 'Forward Jitter (brown)` line is the `SIPSorcery Softphone` audio and the `Reverse Jitter (green)` is the `MicroSIP Softphone` jitter. The `MicroSIP` jitter is substantially flatter (a good thing) and is because it's written in C/C++, an unmanaged memory language, whereas the `SIPSorcery Softphone` is written in C# and .NET Core and has the garbage collection and other memory management overhead. 

![jitter2](https://user-images.githubusercontent.com/197660/90016572-1a2a1680-dca2-11ea-81a4-7dd1bb6dab71.png)

In both cases the softphones are producing a good audio profile. Looking at the audio waveform graph, which is both softphones using the same microphone for audio input, they are essentially identical.

![audio_waveform](https://user-images.githubusercontent.com/197660/90017195-0cc15c00-dca3-11ea-9386-3b6182ddfc43.png)
